docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+|-|-|-|-|
+| systemd | VM root | n/a | The init daemon, running as PID 1 |
+| [Agent](#agent) | VM root | yes | Runs as a systemd service |
+| `chronyd` | VM root | yes | Used to synchronise the time with the host |
+| container workload | VM container | no | Managed by the agent |
+
+See also the [process overview](#process-overview).
 
-The only services running in the context of the mini O/S are the init daemon
-(`systemd`) and the [Agent](#agent). The real workload the user wishes to run
-is created using libcontainer, creating a container in the same manner that is done
-by `runc`.
+> **Note:** The container workload is running inside a full container
+> environment which itself is running within a VM environment.
 
-For example, when `ctr run -ti ubuntu date` is run:
+For example, when the user runs `ctr run -ti ubuntu date`:
 
 - The hypervisor will boot the mini-OS image using the guest kernel.
 - `systemd`, running inside the mini-OS context, will launch the `kata-agent` in
   the same context.
 - The agent will create a new confined context to run the specified command in
   (`date` in this example).
 - The agent will then execute the command (`date` in this example) inside this
-  new context, first setting the root filesystem to the expected Ubuntu\* root
+  new context, first setting the root filesystem to the expected Ubuntu root
   filesystem.
 
 #### Initrd image
 
-A compressed `cpio(1)` archive, created from a rootfs which is loaded into memory and used as part of the Linux startup process. During startup, the kernel unpacks it into a special instance of a `tmpfs` that becomes the initial root filesystem.
+The initrd image is a compressed `cpio(1)` archive, created from a
+rootfs which is loaded into memory and used as part of the Linux
+startup process. During startup, the kernel unpacks it into a special
+instance of a `tmpfs` mount that becomes the initial root filesystem.
+
+The only service running in the context of the initrd is the
+[Agent](#agent) as the init daemon. The real workload the user wishes
+to run is created using libcontainer, creating a container in the same
+manner that is done by `runc`.
+
+| Process | Environment | Notes |
+|-|-|-|
+| [Agent](#agent) | VM root | Runs as the init daemon (PID 1) |
+| container workload | VM container | Managed by the agent |
+
+See also the [process overview](#process-overview).
+
+## Hypervisor
 
-The only service running in the context of the initrd is the [Agent](#agent) as the init daemon. The real workload the user wishes to run is created using libcontainer, creating a container in the same manner that is done by `runc`.
+The [configured](#configuration) [hypervisor](../hypervisors.md) creates
+a VM to host the [agent](#agent) and the [workload](#workload).
+
+> **Note:**
+>
+> The hypervisor process run inside a separate "partial container"
+> environment comprised of cgroups and a separate network namespace
+> from the host.
```

Comment:
I'd also mention that the hypervisor can be SELinux confined, running under the container_kvm_t type (see: https://www.redhat.com/sysadmin/selinux-kata-containers)